File: libraryUse/stackqueue.cpp
#include <iostream>
#include <stack>
#include <queue>
using namespace std;

int main(void) {
  stack<int> stack;
  queue<int> queue;
  priority_queue<int> pri_q;
  priority_queue<int, vector<int>, greater<int>> pri_q2;

  for (int i = 1; i < 10; i++) {
    int val = 3 * i % 10;
    stack.push(val);
    queue.push(val);
    pri_q.push(val);
    pri_q2.push(val);
  }

  cout << "stack:";
  while (!stack.empty()) {
    cout << " " <<  stack.top();
    stack.pop();
  }
  cout << endl;
  cout << "queue:";
  while (!queue.empty()) {
    cout << " " <<  queue.front();
    queue.pop();
  }
  cout << endl;
  cout << "priority queue:";
  while (!pri_q.empty()) {
    cout << " " <<  pri_q.top();
    pri_q.pop();
  }
  cout << endl;
  cout << "priority queue2:";
  while (!pri_q2.empty()) {
    cout << " " << pri_q2.top();
    pri_q2.pop();
  }
  cout << endl;
  return 0;
}

File: libraryUse/stackqueue.py
from collections import deque
from heapq import heappush, heappop


def main():
    stack = deque()
    queue = deque()
    priq = []
    priq2 = []

    for i in range(10):
        val = 3 * i % 10
        stack.append(val)
        queue.append(val)
        heappush(priq, val)
        heappush(priq2, [-val, val])

    print("stack:")
    while stack:
        print(stack.pop())

    print("queue:")
    while queue:
        print(queue.popleft())

    print("priority queue:")
    while priq:
        print(heappop(priq))
    print("priq2")
    while priq2:
        pri, val = heappop(priq2)
        print(val)


main()
